atoma-auth/src/auth.rs

```diff
@@ -131,7 +131,17 @@ pub struct Auth {
 }
 
 impl Auth {
-    /// Constructor
+    /// Constructor for the Auth struct.
+    ///
+    /// # Arguments
+    /// * `config` - Authentication configuration
+    /// * `state_manager_sender` - Channel for sending state manager events
+    /// * `sui` - Sui blockchain client instance
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// - Failed to fetch Google public keys (when google-oauth feature is enabled)
+    #[allow(clippy::unused_async)]
```

Comment:
why `unused_async` ?

Reply:
`google-oauth` is currently disabled but it will be enabled in the future, and when it is then this will become an `async` function, so I think it's best to leave it as such rather than do a mass refactor that will eventually be undone.